File: lcs/lcs.py
#     |
#     |
# X = aaaaaaabbbbbbbbb
# Y = dsfbbbbbbbbdsfaa
#     |
#     |
   
def lcs(x, y):
    lcs = ""
    cs = ""
    i = 0
    j = 0
    k = 0
    n = len(x)
    m = len(y)

    while i < n:
        cs = ""
        j = 0
        k = 0
        print("cs = ", cs)
        print("i = ", i)
        print("j = ", j)
        print("k = ", k)

        while k < m:
            if x[j] == y[k]:
                cs = f"{cs}{y[k]}"

                j += 1

            k += 1
            print("cs = ", cs)
            print("i = ", i)
            print("j = ", j)
            print("k = ", k)

        i += 1

        if len(cs) > len(lcs):
            lcs = cs

        print("lcs ", lcs)

    return lcs   

File: lcs/lcs.c
#include <stdlib.h>
#include <string.h>

static inline int max(int x, int y) {
  return x < y ? y : x;
}

static inline int min(int x, int y) {
  return x < y ? x : y;
}

char *lcs(char *x, char *y) {
  int xi = 0;
  int yi = 0;
  int xj = 0;
  int yj = 0;
  int xl = strlen(x);
  int yl = strlen(y);
  char *cs = (char *)malloc(sizeof(char) * min(xl, yl));
  cs[0] = '\0';
  char *lcs = (char *)malloc(sizeof(char) * min(xl, yl));
  lcs[0] = '\0';

  int si = 0;

traverse_x:
  if (xi < xl) {
    while (xj < xl) {
      if (y[yi] == x[xj]) {
        cs[si++] = x[xj];
        
      }
    } 
  }
traverse_y:
  if (yi < yl) {
    while (yj < xl) {

    }
  }
}
